### PR TITLE
Stm32f4 uF2 prompt the problem of insufficient EEPROM

### PR DESCRIPTION
**Pull request by swkidden**
## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [x] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: platforms/chibios/eeprom_stm32_defs.h [hunk 1]
```diff
@@ -34,7 +34,7 @@
 #        endif
 #    elif defined(STM32F401xC) || defined(STM32F401xE) || defined(STM32F405xG) || defined(STM32F411xE)
 #        ifndef FEE_PAGE_SIZE
-#            define FEE_PAGE_SIZE 0x4000 // Page size = 16KByte
+#            define FEE_PAGE_SIZE 0x4000// Page size = 16KByte
```
Comment (zvecr) [inline platforms/chibios/eeprom_stm32_defs.h:37]:
```suggestion
#            define FEE_PAGE_SIZE 0x4000 // Page size = 16KByte
```

File: builddefs/common_features.mk [hunk 2]
```diff
@@ -183,7 +183,7 @@ else
     ifeq ($(PLATFORM),AVR)
       # Automatically provided by avr-libc, nothing required
     else ifeq ($(PLATFORM),CHIBIOS)
-      ifneq ($(filter STM32F3xx_% STM32F1xx_% %_STM32F401xC %_STM32F401xE %_STM32F405xG %_STM32F411xE %_STM32F072xB %_STM32F042x6 %_GD32VF103xB %_GD32VF103x8, $(MCU_SERIES)_$(MCU_LDSCRIPT)),)
+      ifneq ($(filter STM32F3xx_% STM32F1xx_% %_STM32F401xC%_STM32F401xC_tinyuf2 % %_STM32F401xE %_STM32F405xG %_STM32F411xE %_STM32F411xE_tinyuf2 %_STM32F072xB %_STM32F042x6 %_GD32VF103xB %_GD32VF103x8, $(MCU_SERIES)_$(MCU_LDSCRIPT)),)
```
Comment (zvecr) [inline builddefs/common_features.mk:186]:
```suggestion
      ifneq ($(filter STM32F3xx_% STM32F1xx_% %_STM32F401xC %_STM32F401xC_tinyuf2 %_STM32F401xE %_STM32F405xG %_STM32F411xE %_STM32F411xE_tinyuf2 %_STM32F072xB %_STM32F042x6 %_GD32VF103xB %_GD32VF103x8, $(MCU_SERIES)_$(MCU_LDSCRIPT)),)
```

**Review by zvecr — CHANGES_REQUESTED** [review]
Also <https://github.com/qmk/qmk_firmware/pull/16287#discussion_r822234320> and #16586 for why this change isnt enough on its own.

**Review by sigprof — CHANGES_REQUESTED** [review]
The changes that are done here are incomplete — you also need to change `FEE_PAGE_BASE_ADDRESS` (the default value used on F4x1 would leave only  16 KiB for the bootloader, which is apparently not enough). From https://github.com/qmk/qmk_firmware/pull/16287#discussion_r822234320 it seems that the flash portion reserved for the bootloader code is 32 KiB, and sectors 2 and 3 can be used for EEPROM.

**Comment by zvecr** [comment]
Superseded by #18042

**Pull request closed without merging**